### DISCUSSION TITLE
Automatic map-rotation and campaigns?

### DISCUSSION THREAD

**Discussion by poVoq** [nzp-team/nzportable#1060] [General]
Is there a way to make a server automatically switch maps at the end of a round?

**Comment by poVoq** [comment]
There would be this example code from the FTEQW devs that could maybe be used to add a simple map cycle:
https://code.idtech.space/eukara/quake-qw-qc/commit/d3ad75e188cd80ff8b77095e364a48eb859b466a

(The same repo also contains a simple map change vote through the chat implementation).

**Reply by poVoq** [reply]
Ok, great. So both was implemented and seems to be working great on my dedicated server. Thanks a lot.

**Comment by poVoq** [comment]
Another related idea would be to allow specifying campaigns, i.e. a linear list of maps that follow each other and that tell some sort of story. A bit like Left4Dead does it.

Of course this would first need some sort of victory condition for maps, but otherwise it should be quite similar to the maprotation system that reads the maps from a file. Basically you would just need the option to have multiple such files and then you can easily switch between them for the different campaigns.

**Reply by MotoLegacy** [reply]
This is the purpose of `sv_maprotationbasename`. You can change this cvar on the server to change the rotation file that is loaded. Win conditions already exist in maps via the `func_ending` entity.

**Reply by poVoq** [reply]
Cool, so basically we just need a series of maps that utilize this now.
